lucetc/src/compiler/cpu_features.rs

```diff
@@ -0,0 +1,159 @@
+use crate::error::{LucetcError, LucetcErrorKind};
+use cranelift_codegen::{isa, settings::Configurable};
+use failure::{format_err, ResultExt};
+use target_lexicon::Triple;
+
+/// x86 CPU families used as shorthand for different CPU feature configurations.
+///
+/// Matches the definitions from `cranelift-codegen`'s x86 settings definition.
+#[derive(Debug, Clone, Copy)]
+pub enum TargetCpu {
+    Baseline,
+    Nehalem,
+    Haswell,
+    Broadwell,
+    Skylake,
+    Cannonlake,
+    Icelake,
+    Znver1,
+}
+
+/// A manual specification of the CPU features to use during codegen.
+#[derive(Debug, Clone, Copy)]
+pub struct SpecificFeatures {
+    pub has_sse3: bool,
+    pub has_ssse3: bool,
+    pub has_sse41: bool,
+    pub has_sse42: bool,
+    pub has_popcnt: bool,
+    pub has_avx: bool,
+    pub has_bmi1: bool,
+    pub has_bmi2: bool,
+    pub has_lzcnt: bool,
+}
+
+impl SpecificFeatures {
+    /// Return a `CpuFeatures` with all optional features disabled.
+    pub fn all_disabled() -> Self {
```

Comment:
How about calling this `baseline`? Functionally this means "generate for anything NetBurst/San Diego or newer", but we still require 64-bit support, for example.